instance/tests/integration/integration_instance.py

```diff
@@ -79,6 +81,41 @@ def assert_swift_container_provisioned(self, instance):
         stat_result = stat_container(instance.swift_container_name)
         self.assertEqual(stat_result.read_acl, '.r:*')
 
+    def assert_mysql_db_provisioned(self, instance):
+        """
+        Verify that the MySQL database for the instance has been provisioned and can be
+        connected to with the credentials the instance provides.
+        """
+        for database in instance.mysql_databases:
+            connection = mysql.connect(
+                host=settings.INSTANCE_MYSQL_URL_OBJ.hostname,
+                user=database['user'],
+                passwd=instance._get_mysql_pass(database['user']),
+                port=settings.INSTANCE_MYSQL_URL_OBJ.port or 3306,
+            )
+            database_name = connection.escape_string(database['name']).decode()
+            cur = connection.cursor()
+            result = cur.execute(
+                "SELECT SCHEMA_NAME FROM INFORMATION_SCHEMA.SCHEMATA WHERE SCHEMA_NAME = '{db_name}'".format(
+                    db_name=database_name
+                )
+            )
+            self.assertEqual(result, 1)
+
+    def assert_mongo_db_provisioned(self, instance):
+        """
+        Verify that the databases ostensibly created by the provisioning process actually were.
+        """
+        mongo = pymongo.MongoClient(settings.INSTANCE_MONGO_URL)
+        db_names = mongo.database_names()
+        # This looks a little odd - basically, Mongo's `database_names` command doesn't list
+        # databases that don't have any documents in them. And, we have at least one database
+        # (forums) that doesn't have any documents inserted during the provisioning. So, as a
+        # result, if we require every db_name to be present, we'll fail, even though the db
+        # is, in actuality, there. So, we're checking that at least one of the databases that
+        # we're supposed to create shows up.
```

Comment:
Then a better test may be to just insert a simple document `{"foo": "bar"}` into a database like "temp_test_db"; if that succeeds, everything should be working well, and that test should work even if the assumption I mentioned above changes.